cmd/initializer/internal/config_generator.go

```diff
@@ -123,31 +132,74 @@ func (c *ConfigGenerator) generateDatanodeConfig(initConfig []byte) ([]byte, err
 	return configData, nil
 }
 
+func (c *ConfigGenerator) generateFlownodeConfig(initConfig []byte) ([]byte, error) {
+	cfg, err := dbconfig.NewFromComponentKind(v1alpha1.FlownodeComponentKind)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := cfg.SetInputConfig(string(initConfig)); err != nil {
+		return nil, err
+	}
+
+	flownodeCfg, ok := cfg.(*dbconfig.FlownodeConfig)
+	if !ok {
+		return nil, fmt.Errorf("cfg is not flownode config")
+	}
+
+	nodeID, err := c.allocateNodeID()
+	if err != nil {
+		klog.Fatalf("Allocate node id failed: %v", err)
+	}
+	flownodeCfg.NodeID = &nodeID
+
+	podIP := os.Getenv(deployer.EnvPodIP)
+	if len(podIP) == 0 {
+		return nil, fmt.Errorf("empty pod ip")
+	}
+	flownodeCfg.Addr = util.StringPtr(fmt.Sprintf("%s:%d", podIP, c.RPCPort))
+
+	podName := os.Getenv(deployer.EnvPodName)
+	if len(podName) == 0 {
+		return nil, fmt.Errorf("empty pod name")
+	}
+
+	flownodeCfg.Hostname = util.StringPtr(fmt.Sprintf("%s.%s.%s:%d", podName,
+		c.ServiceName, c.Namespace, c.RPCPort))
+
+	configData, err := dbconfig.Marshal(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return configData, nil
+}
+
 // TODO(zyy17): the algorithm of allocating datanode id will be changed in the future.
-// We use the very easy way to allocate datanode-id: use the pod index of datanode that created by statefulset.
+// We use the very easy way to allocate node id for datanode and flownode: use the pod index of datanode that created by statefulset.
 // If the hostname of datanode is 'basic-datanode-1', then its node-id will be '1'.
-func (c *ConfigGenerator) allocateDatanodeID() (uint64, error) {
+func (c *ConfigGenerator) allocateNodeID() (uint64, error) {
 	name, err := c.hostname()
 	if err != nil {
 		return 0, err
 	}
 
 	if len(name) == 0 {
-		return 0, fmt.Errorf("the hostname is empty")
+		return 0, fmt.Errorf("the datanodeHostname is empty")
```

Comment:
question: The allocateNodeID function contains flownode and datanode, use hostname replace datanodeHostname in error logging ?

Reply:
typo. let me fix it.